### PR TITLE
Implemented transparent handle support

### PR DESCRIPTION
**Pull request by rainman110**
Since handles are now automatically converted in their inheritance chain,
more include files are required. Therefore, a large portion of the
Modules data structure had to be adapted. This could lead to longer
compilation times!

This addresses issue https://github.com/tpaviot/pythonocc-core/issues/539 and PR https://github.com/tpaviot/pythonocc-core/pull/581.

### CODE REVIEW

File: src/Modules.py [hunk 1]
```diff
@@ -271,7 +271,7 @@
            ('TopExp', ['Message', 'TopLoc'], []),
            ('TopTools', [], []),
            ('BRep', ['TShort'], []),
-           ('BRepLProp', [], []),
+           ('BRepLProp', ['GeomAdaptor',  'Geom2dAdaptor', 'Geom', 'Geom2d', 'TopoDS', 'Adaptor3d', 'Adaptor2d', 'TopLoc'], []),
```
Comment (tpaviot) [inline src/Modules.py]: This line is too long

File: src/Modules.py [hunk 2]
```diff
@@ -422,7 +435,9 @@
                          'Adaptor2d', 'GeomAdaptor', 'Geom', 'TopLoc',
                          'Geom2d'],
             ['BRepMesh_WireInterferenceChecker', 'BRepMesh_EdgeTessellator',
-             'BRepMesh_EdgeTessellationExtractor', 'BRepMesh_EdgeParameterProvider' ],
+             'BRepMesh_EdgeTessellationExtractor', 'BRepMesh_EdgeParameterProvider',
+             'BRepMesh_DataStructureOfDelaun', 'BRepMesh_DiscretRoot',
+             'BRepMesh_FastDiscret', 'BRepMesh_FastDiscretFace', 'BRepMesh_IEdgeTool'],
```
Comment (tpaviot) [inline src/Modules.py]: Is it necessary to remove additional classes ? ```BRepMesh_DiscretRoot``` is important, I suggest we don't include this change, if possible

**Comment by tpaviot** [comment]
On my machine, enabling back the BREPMesh classes is working. Here is what I had to do:

```
diff --git a/src/Modules.py b/src/Modules.py
index 65c756f..44590c0 100644
--- a/src/Modules.py
+++ b/src/Modules.py
@@ -435,9 +435,7 @@ OCE_MODULES = [
                          'Adaptor2d', 'GeomAdaptor', 'Geom', 'TopLoc',
                          'Geom2d'],
             ['BRepMesh_WireInterferenceChecker', 'BRepMesh_EdgeTessellator',
-             'BRepMesh_EdgeTessellationExtractor', 'BRepMesh_EdgeParameterProvider',
-             'BRepMesh_DataStructureOfDelaun', 'BRepMesh_DiscretRoot',
-             'BRepMesh_FastDiscret', 'BRepMesh_FastDiscretFace', 'BRepMesh_IEdgeTool'],
+             'BRepMesh_EdgeTessellationExtractor', 'BRepMesh_EdgeParameterProvider'],
             {'BRepMesh_Delaun': ['Frontier', 'InternalEdges', 'FreeEdges'],
              'BRepMesh_GeomTool': ['IntLinLin', 'Normal', 'IntSegSeg'],
              # Both failures caused by Precision::Confusion -> PrecisionConfusion
@@ -485,7 +483,7 @@ OCE_MODULES = [
            ### TKXMesh
            ('XBRepMesh', ['Geom', 'GeomAdaptor', 'Geom2d', 'Bnd', 'TShort',
                           'BRepAdaptor', 'Adaptor2d', 'Geom2dAdaptor', 'TopLoc',
-                          'Poly', 'Adaptor3d'], []),
+                          'Poly', 'Adaptor3d', 'Message', 'TopTools'], []),
            ### TKBool
            ('TopOpeBRep', ['TColStd', 'TCollection', 'TopLoc', 'TColgp',
                            'Adaptor3d', 'IntAna', 'Message', 'IntSurf',
```

**Comment by tpaviot** [comment]
@rainman110 Note: I included the ```Transfer``` change into master, to be sync with pythonocc-core current status. You should rebase this PR onto master to avoid conflicts when merging

File: src/Modules.py (hunk 2 above)
Comment (rainman110) [reply src/Modules.py]: I actually don't know, why this was removed. Is started to work on this using pythonocc 0.17, maybe they were removed there. I will put them back in.

**Comment by rainman110** [comment]
@rainman110 I'll do it. I won't have be the time for it today though.

**Comment by tpaviot** [comment]
Superseeded by PR #47

**Pull request closed without merging**